src/lib.rs

```diff
@@ -46,7 +46,15 @@ pub mod peripheral;
 pub mod peripherals;
 pub mod prelude;
 pub mod reset;
+#[cfg(feature = "rmt-legacy")]
 pub mod rmt;
+//
+#[cfg(all(
+    any(esp32, esp32c3, esp32c6, esp32h2, esp32p4, esp32s2, esp32s3),
```

Comment:
since it works on any esp that has a rmt device, that maybe a better way to include it as to manually list all esp's. That can be done with the flag `esp_idf_soc_rmt_supported`.

Reply:
Curious where all these flags are listed/defined?

Reply:
`cargo rustc -- --print cfg` will show you what is currently defined for your combination of target and sdkconfig's